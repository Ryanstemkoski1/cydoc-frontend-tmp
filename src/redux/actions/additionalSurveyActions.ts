import { UPDATE_ADDITIONAL_DETAILS } from './actionTypes';

export interface UpdateAdditionalSurveyAction {
    type: UPDATE_ADDITIONAL_DETAILS.UPDATE_ADDITIONAL_DETAILS;
    payload: {
        legalFirstName: string;
        legalLastName: string;
        socialSecurityNumber: string;
        dateOfBirth: string;
        showAdditionalSurvey: boolean;
    };
}

export interface GoBackToAdditionalSurvey {
    type: UPDATE_ADDITIONAL_DETAILS.GO_BACK;
    payload: {
        showAdditionalSurvey: boolean;
    };
}

export function updateAdditionalSurveyDetails(
    legalFirstName: string,
    legalLastName: string,
    socialSecurityNumber: string,
    dateOfBirth: string
): UpdateAdditionalSurveyAction {
    return {
        type: UPDATE_ADDITIONAL_DETAILS.UPDATE_ADDITIONAL_DETAILS,
        payload: {
            legalFirstName,
            legalLastName,
            socialSecurityNumber,
            dateOfBirth,
            showAdditionalSurvey: false,
        },
    };
}

export function resetAdditionalSurveyPage(): GoBackToAdditionalSurvey {
    return {
        type: UPDATE_ADDITIONAL_DETAILS.GO_BACK,
        payload: {
            showAdditionalSurvey: true,
        },
    };
}
