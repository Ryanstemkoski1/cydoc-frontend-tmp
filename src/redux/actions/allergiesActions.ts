import { ALLERGIES_ACTION } from './actionTypes';

interface UpdateIncitingAgentAction {
    type: ALLERGIES_ACTION.UPDATE_INCITING_AGENT;
    payload: {
        index: string;
        newIncitingAgent: string;
    };
}

export function updateIncitingAgent(index: string, newIncitingAgent: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_INCITING_AGENT,
        payload: {
            index,
            newIncitingAgent,
        },
    };
}

interface UpdateReactionAction {
    type: ALLERGIES_ACTION.UPDATE_REACTION;
    payload: {
        index: string;
        newReaction: string;
    };
}

export function updateReaction(index: string, newReaction: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_REACTION,
        payload: {
            index,
            newReaction,
        },
    };
}

interface UpdateCommentsAction {
    type: ALLERGIES_ACTION.UPDATE_COMMENTS;
    payload: {
        index: string;
        newComments: string;
    };
}

export function updateComments(index: string, newComments: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_COMMENTS,
        payload: {
            index,
            newComments,
        },
    };
}

interface AddAllergyAction {
    type: ALLERGIES_ACTION.ADD_ALLERGY;
}

export function addAllergy() {
    return {
        type: ALLERGIES_ACTION.ADD_ALLERGY,
    };
}

interface DeleteAllergyAction {
    type: ALLERGIES_ACTION.DELETE_ALLERGY;
    payload: {
        index: string;
    };
}

export function deleteAllergy(index: string) {
    return {
        type: ALLERGIES_ACTION.DELETE_ALLERGY,
        payload: {
            index,
        },
    };
}

export type AllergiesActionTypes =
    | UpdateIncitingAgentAction
    | UpdateReactionAction
    | UpdateCommentsAction
    | AddAllergyAction
    | DeleteAllergyAction;
