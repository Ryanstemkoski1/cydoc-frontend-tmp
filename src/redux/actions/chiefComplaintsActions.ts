import { CHIEF_COMPLAINTS } from './actionTypes';
import { DoctorView } from 'constants/hpiEnums';

export interface SelectChiefComplaintAction {
    type: CHIEF_COMPLAINTS.SELECT_CHIEF_COMPLAINTS;
    payload: {
        disease: DoctorView;
    };
}

export function selectChiefComplaint(
    disease: DoctorView
): SelectChiefComplaintAction {
    return {
        type: CHIEF_COMPLAINTS.SELECT_CHIEF_COMPLAINTS,
        payload: {
            disease,
        },
    };
}

export type chiefComplaintsActionTypes = SelectChiefComplaintAction;
