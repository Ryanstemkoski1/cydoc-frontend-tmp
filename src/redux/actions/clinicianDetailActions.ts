import { initialClinicianDetailType } from 'redux/reducers/clinicianDetailReducer';
import { CLINICIAN_DETAILS } from './actionTypes';

export interface clinicianDetailAction {
    type: CLINICIAN_DETAILS;
    payload: {
        [key: string]: string | number | initialClinicianDetailType;
    };
}

export function setClinicianLastName(last_name: string) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_LAST_NAME,
        payload: {
            last_name,
        },
    };
}

export function setClinicianFirstName(first_name: string) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_FIRST_NAME,
        payload: {
            first_name,
        },
    };
}

export function setClinicianEmail(email: string) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_EMAIL,
        payload: {
            email,
        },
    };
}

export function setClinicianId(id: number) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_ID,
        payload: {
            id,
        },
    };
}

export function setClinicianInstitutionId(institution_id: number) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_ID,
        payload: {
            institution_id,
        },
    };
}
export function setClinicianDetail(obj: initialClinicianDetailType) {
    return {
        type: CLINICIAN_DETAILS.SET_CLINICIAN_DETAIL,
        payload: {
            obj,
        },
    };
}
