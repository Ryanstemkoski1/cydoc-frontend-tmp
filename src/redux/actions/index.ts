import { ROSActionTypes } from './reviewOfSystemsActions';
import { MedicalHistoryActionTypes } from './medicalHistoryActions';
import { SurgicalHistoryActionTypes } from './surgicalHistoryActions';
import { MedicationsActionTypes } from './medicationsActions';
import { AllergiesActionTypes } from './allergiesActions';
import { SocialHistoryActionTypes } from './socialHistoryActions';
import { FamilyHistoryActionTypes } from './familyHistoryActions';
import { PhysicalExamActionTypes } from './physicalExamActions';
import { PlanActionTypes } from './planActions';
import { HpiActionTypes } from './hpiActions';
import { WidgetActionTypes } from './widgetActions';
import { chiefComplaintsActionTypes } from './chiefComplaintsActions';

export type AllActionTypes =
    | ROSActionTypes
    | PhysicalExamActionTypes
    | WidgetActionTypes
    | MedicalHistoryActionTypes
    | SurgicalHistoryActionTypes
    | MedicationsActionTypes
    | AllergiesActionTypes
    | SocialHistoryActionTypes
    | FamilyHistoryActionTypes
    | PlanActionTypes
    | HpiActionTypes
    | chiefComplaintsActionTypes;
