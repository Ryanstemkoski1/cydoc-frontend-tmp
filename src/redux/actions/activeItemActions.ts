import { UnknownAction } from 'redux';
import { ACTIVE_ITEM_ACTION } from './actionTypes';

export interface UpdateActiveItemAction extends UnknownAction {
    type: ACTIVE_ITEM_ACTION.ACTIVE_ITEM;
    payload: {
        updatedItem: string;
    };
}

export function updateActiveItem(updatedItem: string): UpdateActiveItemAction {
    return {
        type: ACTIVE_ITEM_ACTION.ACTIVE_ITEM,
        payload: {
            updatedItem,
        },
    };
}

export type activeItemActionTypes = UpdateActiveItemAction;
