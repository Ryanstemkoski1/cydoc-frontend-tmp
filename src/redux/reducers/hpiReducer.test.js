import { initialHpiState, medId, hpiReducer } from './hpiReducer';
import { HPI_ACTION } from '../actions/actionTypes';
import { BodyLocationOptions, ExpectedResponseDict } from 'constants/hpiEnums';
import { options } from 'pages/EditNote/content/hpi/knowledgegraph/src/components/responseComponents/BodyLocation';

describe('hpi reducers', () => {
    const processedState = {
        graph: {},
        nodes: {
            [medId]: {
                uid: 'uid',
                medID: 'medID',
                category: 'category',
                text: 'text',
                responseType: 'responseType',
                bodySystem: 'bodySystem',
                noteSection: 'noteSection',
                DoctorView: 'DoctorView',
                PatientView: 'PatientView',
                doctorCreated: 'doctorCreated',
                response: 'response',
            },
        },
        edges: {},
    };
    describe('initial state', () => {
        it('returns the initial state', () => {
            const action = { type: 'dummy_action' };
            expect(hpiReducer(undefined, action)).toEqual(initialHpiState);
        });
    });
    describe('add node', () => {
        it('returns node with response', () => {
            const node = {
                uid: 'uid',
                medID: medId,
                category: 'category',
                text: 'text',
                responseType: 'responseType',
                bodySystem: 'bodySystem',
                noteSection: 'noteSection',
                DoctorView: 'DoctorView',
                PatientView: 'PatientView',
                doctorCreated: 'doctorCreated',
                response: 'response',
            };
            const payload = {
                medId: medId,
                node: node,
            };
            const nextState = hpiReducer(initialHpiState, {
                type: HPI_ACTION.ADD_NODE,
                payload,
            });
            expect(nextState).toMatchSnapshot();
            expect(nextState.nodes[medId]).toBeTruthy();
        });
    });
    describe('add edge', () => {
        it('returns correct state with edge info', () => {
            const edge = {
                to: 'foo1',
                from: 'foo2',
                toQuestionOrder: -1,
                fromQuestionOrder: -1,
            };
            const payload = {
                medId: medId,
                edge: edge,
            };
            const nextState = hpiReducer(initialHpiState, {
                type: HPI_ACTION.ADD_EDGE,
                payload,
            });
            expect(nextState).toMatchSnapshot();
            expect(nextState.graph[medId]).toBeTruthy();
            const newEdge = Object.keys(nextState.edges)[0];
            expect(nextState.graph[medId]).toContain(newEdge);
        });
    });
    describe('handle user actions', () => {
        let payload = {};
        let nextState = {};
        beforeEach(() => {
            payload = { medId: medId };
            nextState = processedState;
        });
        it('adds body location responses', () => {
            payload.bodyOptions = options;
            nextState.nodes[medId].responseType = 'BODYLOCATION';
            nextState.nodes[medId].response =
                ExpectedResponseDict['BODYLOCATION'];
            nextState = hpiReducer(nextState, {
                type: HPI_ACTION.BODY_LOCATION_RESPONSE,
                payload,
            });
            expect(nextState).toMatchSnapshot();
        });
        it('handles body location toggle for LRButton', () => {
            nextState.nodes[medId].responseType = 'BODYLOCATION';
            nextState.nodes[medId].response =
                ExpectedResponseDict['BODYLOCATION'];
            nextState = hpiReducer(nextState, {
                type: HPI_ACTION.BODY_LOCATION_RESPONSE,
                payload: {
                    medId: medId,
                    bodyOptions: options,
                },
            });
            payload.bodyOption = BodyLocationOptions.ANKLE;
            payload.toggle = 'left';
            nextState = hpiReducer(nextState, {
                type: HPI_ACTION.BODY_LOCATION_HANDLE_TOGGLE,
                payload,
            });
            expect(nextState).toMatchSnapshot();
            expect(
                nextState.nodes[medId].response[payload.bodyOption][
                    payload.toggle
                ]
            ).toBeTruthy();
            nextState = hpiReducer(nextState, {
                type: HPI_ACTION.BODY_LOCATION_HANDLE_TOGGLE,
                payload,
            });
            expect(
                nextState.nodes[medId].response[payload.bodyOption][
                    payload.toggle
                ]
            ).toBeFalsy();
        });
        it('handles body location toggle for regular button', () => {
            nextState.nodes[medId].responseType = 'BODYLOCATION';
            nextState.nodes[medId].response =
                ExpectedResponseDict['BODYLOCATION'];
            nextState = hpiReducer(nextState, {
                type: HPI_ACTION.BODY_LOCATION_RESPONSE,
                payload: {
                    medId: medId,
                    bodyOptions: options,
                },
            });
            payload.bodyOption = BodyLocationOptions.NOSE;
            payload.toggle = null;
            nextState = hpiReducer(nextState, {
                type: HPI_ACTION.BODY_LOCATION_HANDLE_TOGGLE,
                payload,
            });
            expect(nextState).toMatchSnapshot();
            expect(
                nextState.nodes[medId].response[payload.bodyOption]
            ).toBeTruthy();
            nextState = hpiReducer(nextState, {
                type: HPI_ACTION.BODY_LOCATION_HANDLE_TOGGLE,
                payload,
            });
            expect(
                nextState.nodes[medId].response[payload.bodyOption]
            ).toBeFalsy();
        });
        it('handles multiple choice click', () => {
            payload.name = 'foo';
            nextState.nodes[medId].response = ['foo1'];
            nextState.nodes[medId].responseType = 'CLICK-BOXES';
            expect(
                hpiReducer(nextState, {
                    type: HPI_ACTION.MULTIPLE_CHOICE_HANDLE_CLICK,
                    payload,
                })
            ).toMatchSnapshot();
        });
        it('handles input change', () => {
            payload.textInput = 'foo';
            nextState.nodes[medId].responseType = 'SHORT-TEXT';
            expect(
                hpiReducer(nextState, {
                    type: HPI_ACTION.HANDLE_INPUT_CHANGE,
                    payload,
                })
            ).toMatchSnapshot();
        });
        it('handles numeric input change', () => {
            payload.input = 42;
            nextState.nodes[medId].responseType = 'NUMBER';
            expect(
                hpiReducer(nextState, {
                    type: HPI_ACTION.HANDLE_NUMERIC_INPUT_CHANGE,
                    payload,
                })
            ).toMatchSnapshot();
        });
        describe('handles list text actions', () => {
            const uuid = 'foo';
            const listTextResponse = {
                foo: '',
                1: '1',
                2: '2',
                3: '3',
            };
            beforeEach(() => {
                nextState.nodes[medId].responseType = 'LIST-TEXT';
                nextState.nodes[medId].response = listTextResponse;
            });
            it('handles changing list text input', () => {
                payload.uuid = uuid;
                payload.textInput = 'foo';
                expect(
                    hpiReducer(nextState, {
                        type: HPI_ACTION.LIST_TEXT_HANDLE_CHANGE,
                        payload,
                    })
                ).toMatchSnapshot();
            });
            it('handles removing list text input', () => {
                payload.uuid = uuid;
                expect(
                    hpiReducer(nextState, {
                        type: HPI_ACTION.REMOVE_LIST_INPUT,
                        payload,
                    })
                ).toMatchSnapshot();
            });
            it('handles adding list text input', () => {
                const newResponse = hpiReducer(nextState, {
                    type: HPI_ACTION.ADD_LIST_INPUT,
                    payload,
                }).nodes[medId].response;
                expect(Object.keys(newResponse).length).toEqual(5);
                expect(uuid in newResponse).toBeTruthy();
            });
        });
        describe('handles time input actions', () => {
            beforeEach(() => {
                nextState.nodes[medId].responseType = 'TIME3DAYS';
                nextState.nodes[medId].response = {
                    numInput: 0,
                    timeOption: 'minutes',
                };
            });
            it('handles time input change', () => {
                payload.numInput = 42;
                expect(
                    hpiReducer(nextState, {
                        type: HPI_ACTION.HANDLE_TIME_INPUT_CHANGE,
                        payload,
                    })
                ).toMatchSnapshot();
            });
            it('handles time option change', () => {
                payload.timeOption = 'days';
                expect(
                    hpiReducer(nextState, {
                        type: HPI_ACTION.HANDLE_TIME_OPTION_CHANGE,
                        payload,
                    })
                ).toMatchSnapshot();
            });
        });
        it('handles yes no toggle', () => {
            payload.optionSelected = 'Yes';
            nextState.nodes[medId].responseType = 'YES-NO';
            expect(
                hpiReducer(nextState, {
                    type: HPI_ACTION.YES_NO_TOGGLE_OPTION,
                    payload,
                })
            ).toMatchSnapshot();
        });
        describe('handles scale input actions', () => {
            beforeEach(() => {
                nextState.nodes[medId].responseType = 'SCALE1TO10';
            });
            it('handles scale handle value', () => {
                payload.value = 3;
                expect(
                    hpiReducer(nextState, {
                        type: HPI_ACTION.SCALE_HANDLE_VALUE,
                        payload,
                    })
                ).toMatchSnapshot();
            });
            it('handles scale handle clear', () => {
                expect(
                    hpiReducer(nextState, {
                        type: HPI_ACTION.SCALE_HANDLE_CLEAR,
                        payload,
                    })
                ).toMatchSnapshot();
            });
        });
    });
});
