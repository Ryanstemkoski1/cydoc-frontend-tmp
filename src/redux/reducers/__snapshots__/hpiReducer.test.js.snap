// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hpi reducers add edge returns correct state with edge info 1`] = `
Object {
  "edges": Object {
    "35ae0742-722f-4b8f-92ed-802e2ce9a4b3": Object {
      "from": "foo2",
      "fromQuestionOrder": -1,
      "to": "foo1",
      "toQuestionOrder": -1,
    },
  },
  "graph": Object {
    "medId": Array [
      "35ae0742-722f-4b8f-92ed-802e2ce9a4b3",
    ],
  },
  "nodes": Object {},
}
`;

exports[`hpi reducers add node returns node with response 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "blankNo": undefined,
      "blankTemplate": undefined,
      "blankYes": undefined,
      "response": undefined,
      "responseType": "responseType",
      "text": "text",
    },
  },
}
`;

exports[`hpi reducers handle user actions adds body location responses 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Object {
        "Ankle": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Chest": false,
        "Ear": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Elbow": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Eye": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Finger": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Foot": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Groin": false,
        "Hand": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Head": false,
        "Hip": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Knee": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Lower Arm": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Lower Back": false,
        "Lower Leg": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Mid Back": false,
        "Mouth": false,
        "Neck": false,
        "Nose": false,
        "Pelvis": false,
        "Shoulder": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Stomach": false,
        "Throat": false,
        "Toe": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Upper Arm": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Upper Back": false,
        "Upper Leg": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Wrist": Object {
          "center": false,
          "left": false,
          "right": false,
        },
      },
      "responseType": "BODYLOCATION",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles body location toggle for LRButton 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Object {
        "Ankle": Object {
          "center": false,
          "left": true,
          "right": false,
        },
        "Chest": false,
        "Ear": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Elbow": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Eye": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Finger": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Foot": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Groin": false,
        "Hand": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Head": false,
        "Hip": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Knee": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Lower Arm": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Lower Back": false,
        "Lower Leg": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Mid Back": false,
        "Mouth": false,
        "Neck": false,
        "Nose": false,
        "Pelvis": false,
        "Shoulder": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Stomach": false,
        "Throat": false,
        "Toe": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Upper Arm": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Upper Back": false,
        "Upper Leg": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Wrist": Object {
          "center": false,
          "left": false,
          "right": false,
        },
      },
      "responseType": "BODYLOCATION",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles body location toggle for regular button 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Object {
        "Ankle": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Chest": false,
        "Ear": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Elbow": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Eye": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Finger": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Foot": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Groin": false,
        "Hand": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Head": false,
        "Hip": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Knee": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Lower Arm": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Lower Back": false,
        "Lower Leg": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Mid Back": false,
        "Mouth": false,
        "Neck": false,
        "Nose": true,
        "Pelvis": false,
        "Shoulder": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Stomach": false,
        "Throat": false,
        "Toe": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Upper Arm": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Upper Back": false,
        "Upper Leg": Object {
          "center": false,
          "left": false,
          "right": false,
        },
        "Wrist": Object {
          "center": false,
          "left": false,
          "right": false,
        },
      },
      "responseType": "BODYLOCATION",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles input change 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": "foo",
      "responseType": "SHORT-TEXT",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles list text actions handles changing list text input 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Object {
        "1": "1",
        "2": "2",
        "3": "3",
        "foo": "foo",
      },
      "responseType": "LIST-TEXT",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles list text actions handles removing list text input 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Object {
        "1": "1",
        "2": "2",
        "3": "3",
      },
      "responseType": "LIST-TEXT",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles multiple choice click 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Array [
        "foo1",
        "foo",
      ],
      "responseType": "CLICK-BOXES",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles numeric input change 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": 42,
      "responseType": "NUMBER",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles scale input actions handles scale handle clear 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": undefined,
      "responseType": "SCALE1TO10",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles scale input actions handles scale handle value 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": 3,
      "responseType": "SCALE1TO10",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles time input actions handles time input change 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Object {
        "numInput": 42,
        "timeOption": "minutes",
      },
      "responseType": "TIME3DAYS",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles time input actions handles time option change 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": Object {
        "numInput": 0,
        "timeOption": "days",
      },
      "responseType": "TIME3DAYS",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;

exports[`hpi reducers handle user actions handles yes no toggle 1`] = `
Object {
  "edges": Object {},
  "graph": Object {},
  "nodes": Object {
    "medId": Object {
      "DoctorView": "DoctorView",
      "PatientView": "PatientView",
      "bodySystem": "bodySystem",
      "category": "category",
      "doctorCreated": "doctorCreated",
      "medID": "medID",
      "noteSection": "noteSection",
      "response": "Yes",
      "responseType": "YES-NO",
      "text": "text",
      "uid": "uid",
    },
  },
}
`;
